src/main/scala/com/fasterxml/jackson/module/scala/introspect/ScalaAnnotationIntrospectorModule.scala

```diff
@@ -100,14 +100,15 @@ object ScalaAnnotationIntrospector extends NopAnnotationIntrospector with ValueI
     extends StdValueInstantiator(delegate) {
 
     private val overriddenConstructorArguments: Array[SettableBeanProperty] = {
+      val applyDefaultValues = config.hasDeserializationFeatures(DeserializationFeature.APPLY_DEFAULT_VALUES.getMask)
```

Comment:
this looks wrong - code that checks if a feature is enabled generally looks like this

```
  override def mapArray(jp: JsonParser, ctxt: DeserializationContext): AnyRef = {
    if (ctxt.isEnabled(DeserializationFeature.USE_JAVA_ARRAY_FOR_JSON_ARRAY)) {
```

Reply:
better still use `config: DeserializationConfig` -- it is available where your code change is -- config.isEnabled(...)

Reply:
Switched to `ctxt.isEnabled(DeserializationFeature.APPLY_DEFAULT_VALUES)` I had to move where I did the check to the within the null value provider in order to make it so ObjectReaders could be configured separately.